L1Trigger/TrackFindingTracklet/plugins/ProducerKFout.cc

```diff
@@ -109,19 +197,174 @@ namespace trackFindingTracklet {
       iEvent.getByToken<TTTrackRefMap>(edGetTokenTTTrackRefMap_, handleTTTrackRefMap);
       const TTTrackRefMap& ttTrackRefMap = *handleTTTrackRefMap.product();
-      // perform KFout emulation and fill accepted and lost
-      // StreamsTrack is a vector of StreamTrack which is a vector of FrameTrack which is a pair of an edm::Ref<TTTrack> and std::bitset<64>
-      // the std::bitset<64> are the frames of an emp link
-      // the edm::Ref<TTTrack> is used to meassure tracking efficiency
-      // your input streamsTracks contain edm::Ref<TTTrack> to KF input TTTracks
-      // use ttTrackRefMap to lookup edm::Ref<TTTrack> of KF output TTTracks, that allows us to meassure the tracking efficiency after the KFout block
-      // your output frames belong to either only one TTTrack or to two, in the later case chose any edm::Ref<TTTrack> of the two
-    }
+      // 18 Output Links (First Vector) each has a vector of tracks per event (second vector) each track is 3 32 bit TTBV partial tracks 
+      vector<vector<TTBV>> SortedPartialTracks(setup_->numRegions() * setup_->tfpNumChannel(),vector<TTBV>(0));   
+      StreamsTrack OutputStreamsTracks(setup_->numRegions() * setup_->tfpNumChannel());
+
+      for (int iLink = 0; iLink < (int)streamsTracks.size(); iLink++ ){
+
+        for (int iTrack = 0; iTrack < (int)streamsTracks[iLink].size(); iTrack++ ){
+          auto track = streamsTracks[iLink].at(iTrack);
+
+          std::vector<TTBV> Svalids;
+          std::vector<TTBV> Srs;
+          std::vector<TTBV> Sphis;
+          std::vector<TTBV> Szs;
+          std::vector<TTBV> SdPhis;
+          std::vector<TTBV> SdZs;
+
+          // Create bit vector (gives us all the slicing etc. )
+          TTBV InputLinkTrack(track.second);
+          // Slice Constructor (const TTBV& ttBV, int begin, int end = 0, bool twos =  false)
+          // Unpack KF track 
+          TTBV Tvalid(     InputLinkTrack , wTz + wTcot + wTinvR + wTphi + wTSe + wTSp + wTm + 1, wTz + wTcot + wTinvR + wTphi + wTSe + wTSp + wTm , false );
+          TTBV Tmatch(     InputLinkTrack , wTz + wTcot + wTinvR + wTphi + wTSe + wTSp + wTm    , wTz + wTcot + wTinvR + wTphi + wTSe + wTSp , false );
+          TTBV TsectorPhi( InputLinkTrack , wTz + wTcot + wTinvR + wTphi + wTSe + wTSp          , wTz + wTcot + wTinvR + wTphi + wTSe , false );
+          TTBV TsectorEta( InputLinkTrack , wTz + wTcot + wTinvR + wTphi + wTSe                 , wTz + wTcot + wTinvR + wTphi , false );
+          TTBV TphiT(      InputLinkTrack , wTz + wTcot + wTinvR + wTphi                        , wTz + wTcot + wTinvR , true );
+          TTBV TinvR(      InputLinkTrack , wTz + wTcot + wTinvR                                , wTz + wTcot , true );
+          TTBV Tcot(       InputLinkTrack , wTz + wTcot                                         , wTz , true );
+          TTBV TzT(        InputLinkTrack , wTz                                                 , 0 , true );
+
```

Comment:
Please use this class to unpack KF track: https://github.com/cms-L1TK/cmssw/blob/539dc1a77b060116c34b65a5e371467d44443b8b/L1Trigger/TrackerTFP/interface/DataFormats.h#L649-L650